lib/types/index.d.ts

```diff
@@ -584,14 +584,19 @@ export namespace SourceCode {
 
 // #endregion
 
+type RuleContextNode = {
```

Comment:
I might call this `JSSyntaxElement` instead. That makes it clear that this type represents some kind of syntax rather than some kind of node. (Technically, tokens and comments are not nodes because they aren't part of the tree.)

Reply:
Makes sense, updated 👍🏻